cosmo_tester/framework/test_hosts.py

```diff
@@ -868,6 +870,86 @@ def _set_image_details(self):
                 self._test_config['test_os_usernames'][username_key]
             )
 
+    def prepare_and_dump_xfs(self):
+        self._logger.info('Preparing XFS dump volume for host {}'
+                          .format(self.deployment_id))
+        with self.ssh() as fabric_ssh:
+            # Identify the raw volume
+            volume_id = fabric_ssh.run(
+                "lsblk -o NAME,TYPE -ds | awk '$2 == \"disk\" {print $1}'"
+            ).stdout.strip()
+            if not volume_id:
+                self._logger.warning(
+                    'No raw volume for XFS dump volume found. '
+                    'Proceeding without.')
+                return
+            # Format the volume for XFS
+            fabric_ssh.sudo("mkfs -t xfs /dev/{}".format(volume_id))
+            # Install XFS dump tool
+            fabric_ssh.sudo("yum install -y xfsdump")
+            # Initiate XFS dump
+            self._logger.info(
+                'Creating an XFS dump for host {}. Might take up to 5 '
+                'minutes...'.format(self.deployment_id))
+
+            xfsdump_file = self._tmpdir / 'xfs_dump_{0}'.format(
+                self.ip_address,
+            )
+            xfsdump_file.write_text(
+                "(sudo xfsdump -l0 -L systest -M systest -f "
+                "/dev/{} / && touch /tmp/xfs_dump_complete) "
+                "|| touch /tmp/xfs_dump_failed &".format(volume_id))
+            self.put_remote_file('/tmp/xfs_dump_script', xfsdump_file)
+            fabric_ssh.run('nohup bash /tmp/xfs_dump_script &>/dev/null &')
+            self.xfsdump_volume_id = volume_id
+
+    def restore_xfs(self):
+        with self.ssh() as fabric_ssh:
+            # check whether an XFS restore session exists
+            dump_exists = fabric_ssh.sudo(
+                "xfsrestore -I | grep 'session label:' | grep -q systest "
+                "&& echo 'yup' ||  echo 'nope'").stdout.strip()
+            if dump_exists == "nope":
+                self._logger.error('NO XFS DUMP for host {}!'
+                                   .format(self.deployment_id))
+                raise RuntimeError('XFS restore aborted.')
+            # restore from dump
+            self._logger.info(
+                'Restoring from an XFS dump for host {}. Might take '
+                'up to 1 minute...'.format(self.deployment_id))
+
+            xfsrestore_file = self._tmpdir / 'xfs_restore_{0}'.format(
+                self.ip_address,
+            )
+            xfsrestore_file.write_text(
+                "(sudo xfsrestore -L systest -f /dev/{} / "
+                "&& touch /tmp/xfs_restore_complete) "
+                "|| touch /tmp/xfs_restore_failed &".format(
+                 self.xfsdump_volume_id))
+            self.put_remote_file('/tmp/xfs_restore_script', xfsrestore_file)
+            fabric_ssh.run('nohup bash /tmp/xfs_restore_script &>/dev/null &')
+
+    def is_complete(self, process_name=None):
```

Comment:
Might be better called async_command_is_complete, and with command_name (/process_name) required.